scripts/switching_parallel.py

```diff
@@ -24,62 +29,81 @@
+qml_to_mm_filename = f"{w_dir}/switching_{ff}/{name}_neq_ws_from_qml_to_mm_{nr_of_switches}_{switching_length}_{run_id}.pickle"
 
 if path.isfile(mm_to_qml_filename) and path.isfile(qml_to_mm_filename):
     sys.exit()
 
+# create folder
+os.makedirs(f"{w_dir}/switching_{ff}", exist_ok=True)
+print(f"Generate directory: {w_dir}/switching_{ff}")
+
 ###########################################################################################
 ###########################################################################################
 # generate mol
-molecule = generate_molecule(smiles)
-sim = initialize_simulation(molecule, w_dir=w_dir)
+if ff == "openff" and smiles:
+    molecule = generate_molecule(forcefield=ff, smiles=smiles)
+elif ff == "charmmff" and smiles and not name:
+    raise RuntimeError("Charmff can not be used with SMILES input")
+else:
+    molecule = generate_molecule(forcefield=ff, name=name, base="../data/hipen_data")
+
+# initialize simulation depending on ff keyword
+if ff == "openff":
+    sim = initialize_simulation_with_openff(
+        molecule=molecule, w_dir=f"/data/shared/projects/endstate_rew/{name}/"
+    )
+elif ff == "charmmff":
+    sim = initialize_simulation_with_charmmff(molecule=molecule, zinc_id=name)
 ###########################################################################################
-# create folder
-os.makedirs(f"{w_dir}/switching/{run}", exist_ok=True)
-print(f"Generate directory: {w_dir}/switching/{run}")
 
 # load samples
-mm_samples = pickle.load(
-    open(
-        f"{w_dir}/sampling/{run}/{name}_mm_samples_{n_samples}_{n_steps_per_sample}.pickle",
-        "rb",
-    )
+mm_samples = []
+mm_sample_files = glob(
+    f"/data/shared/projects/endstate_rew/{name}/sampling_{ff}/run*/{name}_samples_{n_samples}_steps_{n_steps_per_sample}_lamb_0.0000.pickle"
 )
-qml_samples = pickle.load(
-    open(
-        f"{w_dir}/sampling/{run}/{name}_qml_samples_{n_samples}_{n_steps_per_sample}.pickle",
-        "rb",
-    )
+for samples in mm_sample_files:
+    mm_samples.append(pickle.load(open(samples, "rb")))
+
+qml_samples = []
+qml_sample_files = glob(
+    f"/data/shared/projects/endstate_rew/{name}/sampling_{ff}/run*/{name}_samples_{n_samples}_steps_{n_steps_per_sample}_lamb_1.0000.pickle"
 )
+for samples in qml_sample_files:
+    qml_samples.append(pickle.load(open(samples, "rb")))
 
 # perform switching only if file does not already exist
 if not path.isfile(mm_to_qml_filename):
     # define lambda space
     lambs = np.linspace(0, 1, switching_length)
     # perform NEQ from MM to QML
-    ws_from_mm_to_qml = perform_switching(
-        sim, lambdas=lambs, samples=mm_samples, nr_of_switches=nr_of_switches
-    )
+    ws_from_mm_to_qml = []
+    for sample in mm_samples:
+        ws_from_mm_to_qml.append(
+            perform_switching(
+                sim, lambdas=lambs, samples=sample, nr_of_switches=nr_of_switches
+            )
+        )
```

Comment:
I am unsure about these lines. It appears that here we perform switching for each `sample` in `mm_samples`/`qml_samles`. That would be not ideal, since we want to randomly select a subset from `mm_samples`/`qml_samles`. As far as I remember, this is done inside the `perform_swtiching` function --- there, we randomly select a number of samples (defined by `nr_of_switches`) from `samples`.

I think these lines can stay exactly the same as before:
`    ws_from_mm_to_qml = perform_switching(
        sim, lambdas=lambs, samples=mm_samples, nr_of_switches=nr_of_switches
    )
`